### PR TITLE
[ceph_mgr] Add more data collection to ceph_mgr

### PR DESCRIPTION
**Pull request by nkshirsagar**
Enhance the ceph_mgr plugin to collect more data specific to
mgr nodes. Also collect some additional manager data in the
ceph_mon plugin.

Related: #1945
Resolves: #2759

Signed-off-by: Nikhil Kshirsagar <nkshirsagar@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/ceph_mgr.py [hunk 1]
```diff
@@ -40,16 +40,26 @@ def setup(self):
             "ceph mgr metadata",
         ])
 
-        # more commands to be added later
-        ceph_cmds = [
-            "mgr module ls",
-            "mgr dump",
+        ceph_d_cmds = [
```
Comment (pponnuvel) [inline sos/report/plugins/ceph_mgr.py:42]: `ceph_d_cmds`? Did you mean to say `ceph_cmds` or `ceph_mgr_cmds` or something like that?

File: sos/report/plugins/ceph_mgr.py [hunk 2]
```diff
@@ -40,16 +40,26 @@ def setup(self):
...
+            "config diff",
+            "config show",
+            "dump_cache",
+            "dump_mempools",
+            "dump_osd_network",
+            "mds_requests",
+            "mds_sessions",
+            "objecter_requests",
+            "get_command_descriptions",
```
Comment (pponnuvel) [inline sos/report/plugins/ceph_mgr.py]: Remove `get_command_descriptions`?

File: sos/report/plugins/ceph_mon.py [hunk 3]
```diff
@@ -51,6 +51,11 @@ def setup(self):
             "ceph osd erasure-code-profile ls",
             "ceph osd crush show-tunables",
             "ceph osd crush dump"
+            "ceph mgr dump",
```
Comment (pponnuvel) [inline sos/report/plugins/ceph_mon.py:54]: Maybe add a note why these are here instead of ceph mgr.

File: sos/report/plugins/ceph_mgr.py (hunk 1 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mgr.py:42]: using ceph_d_cmds for ceph_daemon commands list.

File: sos/report/plugins/ceph_mgr.py (hunk 2 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mgr.py]: ack, done.

File: sos/report/plugins/ceph_mon.py (hunk 3 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mon.py:54]: ack

File: sos/report/plugins/ceph_mgr.py (hunk 1 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mgr.py:42]: changed it the way you suggested on IRC - i.e,

                    containers_list[0][1], "ceph daemon %s %s" % (mgrid, cmd))
                    for mgrid in mgr_ids for cmd in ceph_daemon_cmds

File: sos/report/plugins/ceph_mon.py [hunk 4]
```diff
@@ -50,7 +53,12 @@ def setup(self):
             "ceph osd metadata",
             "ceph osd erasure-code-profile ls",
             "ceph osd crush show-tunables",
-            "ceph osd crush dump"
+            "ceph osd crush dump",
+            "ceph mgr dump",
+            "ceph mgr metadata",
+            "ceph mgr module ls",
+            "ceph mgr services",
+            "ceph mgr versions"
```
Comment (TurboTurtle) [inline sos/report/plugins/ceph_mon.py:61]: This seems very strange to me to move `mgr` commands _out_ of the `ceph_mgr` plugin, even with the note provided. It is something I think users would certainly not expect at the least.

Comment (pponnuvel) [reply sos/report/plugins/ceph_mon.py:61]: @TurboTurtle It's really being taken "out" of ceph-mgr plugun - previously it was all part of the single ceph plugin where each cmd was every unit and and we had all the info w/ sosreports from all units. Now that the plugin has been split, only the relevant commands are run on given nodes. This is all fine.

But the `ceph mgr ...` commands require admin keyring which may not be present on mgr units. If so, we would lose out `ceph mgr ...` data.

I agree that this sounds odd to say `ceph mgr ...` commands may not work on a ceph-mgr node. But this is probably a historic as ceph-mgr was initially part of the ceph-mon daemon and why these commands don't work with just ceph-mgr keyring.

The bottom line is if we kept the `ceph mgr ...` commands in ceph-mgr plugin, these commands may not be a part of *any* sosreport unless ceph-mgr happens to have the admin keyring. So we might miss out those data completely.

Comment (TurboTurtle) [reply sos/report/plugins/ceph_mon.py:61]: Ack. In the end, the folks who use this data will, more likely than not, understand the apparent schism here.

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]
